include/fw_info.h

```diff
@@ -214,41 +277,48 @@ static inline const struct fw_info *fw_info_check(u32_t fw_info_addr)
 
 
 /* The supported offsets for the fw_info struct. */
-#define FW_INFO_OFFSET0 0x200
-#define FW_INFO_OFFSET1 0x400
-#define FW_INFO_OFFSET2 0x800
-#define FW_INFO_OFFSET_COUNT 3
+#define FW_INFO_OFFSET0 0x0
+#define FW_INFO_OFFSET1 0x200
+#define FW_INFO_OFFSET2 0x400
+#define FW_INFO_OFFSET3 0x800
+#define FW_INFO_OFFSET4 0x1000
+#define FW_INFO_OFFSET_COUNT 5
 
 /* Find the difference between the start of the current image and the address
  * from which the firmware info offset is calculated.
  */
 #if defined(PM_S0_PAD_SIZE) && (PM_ADDRESS == PM_S0_IMAGE_ADDRESS)
-	#define VECTOR_OFFSET PM_S0_PAD_SIZE
+	#define FW_INFO_VECTOR_OFFSET PM_S0_PAD_SIZE
 #elif defined(PM_S1_PAD_SIZE) && (PM_ADDRESS == PM_S1_IMAGE_ADDRESS)
-	#define VECTOR_OFFSET PM_S1_PAD_SIZE
+	#define FW_INFO_VECTOR_OFFSET PM_S1_PAD_SIZE
 #elif defined(PM_MCUBOOT_PAD_SIZE) && \
 		(PM_ADDRESS == PM_MCUBOOT_PRIMARY_APP_ADDRESS)
-	#define VECTOR_OFFSET PM_MCUBOOT_PAD_SIZE
+	#define FW_INFO_VECTOR_OFFSET PM_MCUBOOT_PAD_SIZE
 #else
-	#define VECTOR_OFFSET 0
+	#define FW_INFO_VECTOR_OFFSET 0
 #endif
 
-#define CURRENT_OFFSET (CONFIG_FW_INFO_OFFSET + VECTOR_OFFSET)
+#define FW_INFO_CURRENT_OFFSET (CONFIG_FW_INFO_OFFSET + FW_INFO_VECTOR_OFFSET)
 
-static const u32_t allowed_offsets[] = {FW_INFO_OFFSET0, FW_INFO_OFFSET1,
-					FW_INFO_OFFSET2};
+static const u32_t fw_info_allowed_offsets[] = {
+					FW_INFO_OFFSET0, FW_INFO_OFFSET1,
+					FW_INFO_OFFSET2, FW_INFO_OFFSET3,
+					FW_INFO_OFFSET4};
```

Comment:
If the FW_INFO_OFFSET is always increasing array, what about making use of a FW_INFO_OFFSET_INVALID_LAST at the end? This would simplify the logic for the count further down.